llvm/lib/Transforms/Yk/ControlPoint.cpp

```diff
@@ -88,36 +89,149 @@ CallInst *findControlPointCall(Module &M) {
   return cast<CallInst>(*U);
 }
 
-/// Extract all live variables that need to be passed into the control point.
+/// Wrapper to make `std::set_difference` more concise.
 ///
-/// YKFIXME: This currently computes an over-approximation of what's live.
-/// https://github.com/ykjit/yk/issues/515
-std::vector<Value *> getLiveVars(DominatorTree &DT, CallInst *OldCtrlPoint) {
-  std::vector<Value *> Vec;
-  Function *Func = OldCtrlPoint->getFunction();
-
-  // Add function arguments to the live set.
-  for (Value &Arg : Func->args()) {
-    Vec.push_back(&Arg);
+/// Store the difference between `S1` and `S2` into `Into`.
+void vset_difference(const std::set<Value *> &S1, const std::set<Value *> &S2,
+                     std::set<Value *> &Into) {
+  std::set_difference(S1.begin(), S1.end(), S2.begin(), S2.end(),
+                      std::inserter(Into, Into.begin()));
+}
+
+/// Wrapper to make `std::set_union` more concise.
+///
+/// Store the union of `S1` and `S2` into `Into`.
+void vset_union(const std::set<Value *> &S1, const std::set<Value *> &S2,
+                std::set<Value *> &Into) {
+  std::set_union(S1.begin(), S1.end(), S2.begin(), S2.end(),
+                 std::inserter(Into, Into.begin()));
+}
+
+namespace llvm {
+
+// A liveness analysis for LLVM IR.
+//
+// This based on the algorithm shown in Chapter 10 of the book:
```

Comment:
s/This/This is/

Reply:
Fixed in d6bd98977f8e36052d69957799b5addfc1f7a96c